### PR TITLE
DOC: Document core mechanics of pypdf

### PR DESCRIPTION
**Pull request by MartinThoma**
I want to document how pypdf reads/writes PDF files

## For reviewers

* The [rendered docs after this PR](https://pypdf--1783.org.readthedocs.build/en/1783/dev/pypdf-writing.html)

### CODE REVIEW

**Comment by MartinThoma** [comment]
@pubpub-zz I think I'm happy now. Could you please check if it is correct / makes sense?

**Review by pubpub-zz — APPROVED** [review]
This sounds good to me.
Perhaps adding some reference to the pdf ref (1.7? 2.0?)

**Comment by pubpub-zz** [comment]
Schön 😊

**Review by Lucas-C — APPROVED** [review]
Nice documentation addition!

I don't know `pypdf` enough yet to make many constructive suggestions,
but elements could be added later on,
and it's definitively a good start!

File: docs/dev/pypdf-parsing.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# How pypdf parses PDF files
+
+pypdf uses {py:class}`PdfReader <pypdf.PdfReader>` to parse PDF files.
+The method {py:method}`PdfReader.reader <pypdf.PdfReader.reader>` shows the basic
```
[the remark this answers is not in the document]
Comment (MartinThoma) [reply docs/dev/pypdf-parsing.md]: Thank you for mentioning it - I fixed it :-)

**Pull request merged**